src/main/java/com/liefery/android/gallery/GalleryView.java

```diff
@@ -147,11 +151,20 @@ private void initialize( @NonNull TypedArray styles ) {
             photoAuxilery.setGalleryView( this );
         }
 
-        addButton();
+        addPhotoButton = addButton();
 
         styles.recycle();
     }
 
+    public void setMaxPhotos( int maxPhotos ) {
+        this.maxPhotos = maxPhotos;
+
+        if ( addPhotoButton != null && getImages().size() >= maxPhotos )
+            addPhotoButton.setVisibility( View.GONE );
+        else if ( addPhotoButton != null )
+            addPhotoButton.setVisibility( View.VISIBLE );
```

Comment:
How about?
```java
if ( addPhotoButton != null )
    addPhotoButton.setVisibility( getImages().size() >= maxPhotos ? View.GONE : View.VISIBLE );
```

Reply:
Or we could also extract this to a function a call it here, in `addPhoto` and `removePhoto`.

Reply:
I have no preference between ternary and if else, but I updated my code accordingly. Thanks for the review! 🚀